express_relay/sdk/js/src/examples/simpleSearcherSvm.ts

```diff
@@ -0,0 +1,182 @@
+const DUMMY_PIDS: Record<string, PublicKey> = {
+  solana: new PublicKey("HYCgALnu6CM2gkQVopa1HGaNf8Vzbs9bomWRiKP267P3"),
+};
+
+class SimpleSearcherSvm {
+  private client: Client;
+  private connectionSvm: Connection;
+  constructor(
+    public endpointExpressRelay: string,
+    public chainId: string,
+    public privateKey: string,
+    public endpointSvm: string,
+    public apiKey?: string
+  ) {
+    this.client = new Client(
+      {
+        baseUrl: endpointExpressRelay,
+        apiKey,
+      },
+      undefined,
+      () => {
+        return Promise.resolve();
+      },
+      this.bidStatusHandler.bind(this)
+    );
+    this.connectionSvm = new Connection(endpointSvm, "confirmed");
+  }
+
+  async bidStatusHandler(bidStatus: BidStatusUpdate) {
+    let resultDetails = "";
+    if (bidStatus.type == "submitted" || bidStatus.type == "won") {
+      resultDetails = `, transaction ${bidStatus.result}`;
+    } else if (bidStatus.type == "lost") {
+      if (bidStatus.result) {
+        resultDetails = `, transaction ${bidStatus.result}`;
+      }
+    }
+    console.log(
+      `Bid status for bid ${bidStatus.id}: ${bidStatus.type.replaceAll(
+        "_",
+        " "
+      )}${resultDetails}`
+    );
+  }
+
+  async dummyBid() {
+    const secretKey = anchor.utils.bytes.bs58.decode(this.privateKey);
+    const searcher = Keypair.fromSecretKey(secretKey);
+
+    const provider = new AnchorProvider(
+      this.connectionSvm,
+      new anchor.Wallet(searcher),
+      {}
+    );
+    const dummy = new Program<Dummy>(dummyIdl as Dummy, provider);
+
+    const permission = PublicKey.default;
+    const bidAmount = new anchor.BN(argv.bid);
+
+    const svmConstants = SVM_CONSTANTS[this.chainId];
+    if (!(this.chainId in DUMMY_PIDS)) {
+      throw new Error(`Dummy program id not found for chain ${this.chainId}`);
+    }
+    const dummyPid = DUMMY_PIDS[this.chainId];
+    const dummyAccountInfo = await this.connectionSvm.getAccountInfo(dummyPid);
+    let dummyExecutable = false;
+    if (dummyAccountInfo) {
+      dummyExecutable = dummyAccountInfo.executable;
+    }
+
+    const ixDummy = await dummy.methods
+      .doNothing()
+      .accountsPartial({
+        payer: searcher.publicKey,
+        expressRelay: svmConstants.expressRelayProgram,
+        sysvarInstructions: anchor.web3.SYSVAR_INSTRUCTIONS_PUBKEY,
+        permission,
+        protocol: dummyPid,
+      })
+      .instruction();
+    ixDummy.programId = dummyPid;
+
+    const txRaw = new anchor.web3.Transaction().add(ixDummy);
+
+    const bid = await this.client.constructSvmBid(
+      txRaw,
+      searcher.publicKey,
+      dummy.programId,
+      dummyExecutable,
+      permission,
+      bidAmount,
+      new anchor.BN(Math.round(Date.now() / 1000 + DAY_IN_SECONDS)),
+      this.chainId
+    );
+
+    try {
+      const { blockhash } = await this.connectionSvm.getLatestBlockhash();
+      bid.transaction.recentBlockhash = blockhash;
```

Comment:
I would rather have the signing here and outside the sdk. Similar to the RPC argument, I can see searchers implementing a bunch of logging and monitoring around the signing procedures and not willing to pass the underlying keypair around so much